c10/core/CPUAllocator.cpp

```diff
@@ -0,0 +1,74 @@
+#include <c10/core/CPUAllocator.h>
+#include <c10/util/typeid.h>
+#include <c10/DeviceType.h>
+
+C10_DEFINE_bool(
+    caffe2_report_cpu_memory_usage,
+    false,
+    "If set, print out detailed memory usage");
+
+C10_DEFINE_bool(
+    caffe2_cpu_allocator_do_zero_fill,
+    true,
+    "If set, do memory zerofilling when allocating on CPU");
+
+C10_DEFINE_bool(
+    caffe2_cpu_allocator_do_junk_fill,
+    false,
+    "If set, fill memory with deterministic junk when allocating on CPU");
+
+namespace c10 {
+
+void memset_junk(void* data, size_t num) {
+  // This garbage pattern is NaN when interpretted as floating point values,
+  // or as very large integer values.
+  static constexpr int32_t kJunkPattern = 0x7fedbeef;
+  static constexpr int64_t kJunkPattern64 =
+      static_cast<int64_t>(kJunkPattern) << 32 | kJunkPattern;
+  int32_t int64_count = num / sizeof(kJunkPattern64);
+  int32_t remaining_bytes = num % sizeof(kJunkPattern64);
+  int64_t* data_i64 = reinterpret_cast<int64_t*>(data);
```

Comment:
FYI: This is a strict aliasing violation. You're supposed to use a `char*` pointer in this situation and use memcpy to copy the integers over.

Reply:
cc @zheng-xq, original author of code